### PR TITLE
A couple of ntp metrics

### PR DESCRIPTION
**Pull request by dmuino**
Add `sys.time.offset` and `sys.time.lastSampleAge` to the set of metrics
generated by the system agent.

`sys.time.offset`: difference between the system time and what chrony
estimates the clock should be (the difference is there to avoid stepping
the system clock)

`sys.time.lastSampleAge`: number of seconds since the last sample was
received from the ntp source

### CODE REVIEW

File: lib/files.h [hunk 1]
```diff
@@ -30,6 +30,32 @@ class StdIoFile {
   char buf[65536];
 };
 
+class StdPipe {
+ public:
+  explicit StdPipe(const char* cmd) noexcept : fp_(::popen(cmd, "r")) {
+    if (fp_ == nullptr) {
+      Logger()->warn("Unable to execute {}", cmd);
```
Comment (brharrington) [inline lib/files.h]: Is chrony used in all of our supported environments? I'm wondering how noisy this would be.

File: lib/chrony.cc [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        sysLastRx_->Set(last_rx);
+      } catch (const std::invalid_argument& e) {
+        Logger()->error("Unable to parse {} as a double: {}", source_fields[6], e.what());
+      }
+    }
+  }
+}
+
+void Chrony::update_stats() noexcept {
+  auto tracking_csv = read_output_string("chronyc -c tracking");
```
Comment (brharrington) [inline lib/chrony.cc]: Will `chronyc` always be on the path?

Comment (dmuino) [reply lib/chrony.cc]: On our environments `chronyc` is under `/usr/bin/chronyc`, so given our start up there that is correct.

File: lib/files.h (hunk 1 above)
Comment (dmuino) [reply lib/files.h]: Good question. I was considering adding a dependency to our build that generates the package to ensure this, or fail the installation.

**Pull request merged**